main.rb

```diff
@@ -0,0 +1,152 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    array = to_a
+    index = 0
+    while index < size
+      yield(array[index])
+      index += 1
+    end
+    array
+  end
```

Comment:
```rb
2.6.5 :014 > block = proc { |num| num < 4 }
2.6.5 :015 > range = Range.new(5, 50)
2.6.5 :016 > range.my_each(&block)
 => [5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50] 
2.6.5 :017 > range.each(&block)
 => 5..50 
2.6.5 :018 > 
```

Reply:
I have checked the requirements and this wasn't stated for my_each method and I have clearly checked other projects to see how they did theirs of which they used the same method I used, this is you just saying start afresh and it is something I will not do.